enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+---
+title: aws-efa-networking
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - "@elmiko"
+  - "@deads2k"
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+  - "@deads2k"
+creation-date: 2021-12-15
+last-updated: 2021-12-15
+tracking-link:
+  - https://issues.redhat.com/browse/OCPCLOUD-1353
+---
+
+# AWS EFA Networking
+
+## Summary
+
+AWS Elastic-Fabric-Adapter (EFA) is a feature within EC2 that allows for high throughput network adapters to be
+attached to certain instance types. With the presence of these network adapters and some additional drivers,
+workloads can take advantage of greatly improved network performance between nodes within the same AWS availability
+zone.
+
+## Motivation
+
+We have customers already using this feature of EC2 within OpenShift UPI clusters. They have taken on the responsibility
+of making sure the prerequisites are fulfilled and have manually created instances with the attached adapters.
+However, there is a desire to enable this feature to be used in conjunction with existing autoscaling features within
+OpenShift. To enable this, we must provide support for the feature within the Machine API.
+
+### Goals
+
+- Enable worker Machines to attach EFA network interfaces as their primary network interface
+- Enable autoscaling of Machines that leverage high performance network interfaces
+
+### Non-Goals
+
+- Support and or configuration of the prerequisite requirements for using EFA
```

Comment:
Support and/or configuration for any other HPC-related features in AWS (eg, placing the HPC VMs into a particular cluster placement group)